dagwriter.go

```diff
@@ -0,0 +1,79 @@
+package dagwriter
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+
+	blocks "github.com/ipfs/go-block-format"
+	blockstore "github.com/ipfs/go-ipfs-blockstore"
+	"github.com/ipld/go-ipld-prime"
+	cidlink "github.com/ipld/go-ipld-prime/linking/cid"
+)
+
+type DagWriter interface {
+	Store(lnkCtx ipld.LinkContext, lp ipld.LinkPrototype, n ipld.Node) (ipld.Link, error)
+	Delete(ipld.Link) error
```

Comment:
These should take contexts. Eventually, the blockstore should take contexts as well but we need to be able to cancel remote/slow operations.

Reply:
Unless the dag writer is some "ephemeral" thing and we're supposed to to construct it with a context? But I'd prefer not to do that.

Reply:
LinkContext has a Context inside.

Delete doesn't have have a context in Blockstore / BlockService -- should I give it one.

Reply:
> LinkContext has a Context inside.

https://github.com/ipld/go-ipld-prime/issues/154

> Delete doesn't have have a context in Blockstore / BlockService -- should I give it one?

Yes. The plan has been to slowly pipe the context all the way down into the datastore. If for nothing else, metrics.

Same with any flush/commit methods.